### PR TITLE
[ENG-846] Preferences

### PR DESCRIPTION
**Pull request by Brendonovich**

### CODE REVIEW

File: core/src/sync/manager.rs [hunk 1]
```diff
@@ -303,6 +303,7 @@ impl SyncManager {
 						.await?;
 				}
 			},
+			ModelSyncData::Preference(_, _) => todo!(),
```
Comment (brxken128) [inline core/src/sync/manager.rs:306]: 😠

**Review by brxken128 — APPROVED** [review]
Looks good!

I saw some `serde`, but could we maybe get `bincode ^2.0.0-rc.3`  support added at some point? Doesn't have to be in this PR, but I tend to use `bincode` quite a lot and it'd definitely help.

Additionally, what's stopping direct usage of `db.preference()`? Just a curiosity, if needs be we can keep an eye out for direct usage within future PR reviews.

File: core/src/sync/manager.rs (hunk 1 above)
Comment (Brendonovich) [reply core/src/sync/manager.rs:306]: dw about this i'll fix it in due time haha

**Comment by Brendonovich** [comment]
> I saw some serde, but could we maybe get bincode ^2.0.0-rc.3 support added at some point? Doesn't have to be in this PR, but I tend to use bincode quite a lot and it'd definitely help.

I'm using msgpack for this and would rather it over bincode, am gonna be using msgpack in the sync system eventually too.

> Additionally, what's stopping direct usage of db.preference()? Just a curiosity, if needs be we can keep an eye out for direct usage within future PR reviews.

Type safety of preference keys + DX when creating preferences

**Review by oscartbeaumont — APPROVED** [review]

**Pull request merged**